modules/dhp_connected_devices/lib/fitbit/client.rb

```diff
@@ -6,34 +6,50 @@
 
 module DhpConnectedDevices
   module Fitbit
+    class MissingAuthError < StandardError; end
+    class TokenExchangeError < StandardError; end
+
     class Client < Common::Client::Base
       configuration DhpConnectedDevices::Fitbit::Configuration
-
       ##
-      # HTTP POST call to the Fitbit API to exchange auth code for a user token
-      #
-      # @return [Faraday::Response]
+      # HTTP POST call to the Fitbit API to exchange a string of the auth code for a user token
       #
+      # @return [Hash]
       def get_token(auth_code)
-        connection.post(config.base_path) do |req|
+        resp = connection.post(config.base_path) do |req|
           req.headers = headers
           req.body = "client_id=#{Settings.dhp.fitbit.client_id}" \
                      "&code=#{auth_code}" \
                      "&code_verifier=#{CODE_VERIFIER}" \
                      '&grant_type=authorization_code' \
                      "&redirect_uri=#{Settings.dhp.fitbit.redirect_uri}"
         end
+
+        raise "response code: #{resp.status}" unless resp.status == 200
```

Comment:
mentioned it previously, but probably worth including the resp body as well here for more detail about what happened